kytos/core/auth.py

```diff
@@ -44,6 +50,83 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+class UserController:
+    """UserController"""
+
+    # pylint: disable=unnecessary-lambda
+    def __init__(self, get_mongo=lambda: Mongo()):
+        self.mongo = get_mongo()
+        self.db_client = self.mongo.client
+        self.db = self.db_client[self.mongo.db_name]
+
+    def bootstrap_indexes(self):
+        """Bootstrap all topology related indexes."""
+        index_tuples = [
+            ("users", [("username", pymongo.ASCENDING)]),
+        ]
+        for collection, keys in index_tuples:
+            if self.mongo.bootstrap_index(collection, keys):
+                LOG.info(
+                    f"Created DB index {keys}, collection: {collection}"
+                )
+
+    def create_user(self, user_data):
+        """Create user to database"""
+        utc_now = datetime.datetime.utcnow()
+        try:
+            data = {
+                "username": user_data["username"],
+                "email": user_data["email"],
+                "password": user_data["password"],
+                "inserted_at": utc_now,
+                "updated_at": utc_now,
+            }
+            user = UserDoc(**data)
+            result = self.db.users.insert_one(user.dict())
+        except (ValueError, KeyError, Exception) as e:
+            raise e
+        return result
+
+    def delete_user(self, username):
+        """Delete user from database"""
+        utc_now = datetime.datetime.utcnow()
+        update = {
+                "state": "inactive",
+                "deleted_at": utc_now,
+        }
+        result = self.db.users.find_one_and_update(
+            {"username": username},
+            {"$set": update},
+            return_document=ReturnDocument.AFTER,
+        )
+        return result
+
+    def update_user(self, username, data):
+        """Update user from database"""
+        data.update({"updated_at": datetime.datetime.utcnow()})
+        result = self.db.users.find_one_and_update(
+            {"username": username},
+            {"$set": data},
```

Comment:
When trying to update an user we need similar validations from pydantic like we have for `UserDoc`, maybe you can create a `UserDocUpdate` that has `password: Optional[constr(min_length=8)], email: Optional[EmailStr]`, and then when getting the dict you pass `exclude_none=True`. The caller handling a request will be able to try and catch a `pydantic.ValidationError`. 

We haven't completely specified the response for errors in an API spec (we'll do it in the future), but as long as the error information is there we can keep enhancing this and standardizing.

Reply:
Good idea, updated.

Reply:
`UserDocUpdate` added.